corehq/apps/accounting/payment_handlers.py

```diff
@@ -102,10 +104,18 @@ def process_request(self, request):
                     return {'success': True, 'removedCard': card, }
                 if save_card:
                     card = self.payment_method.create_card(card, billing_account, self.domain, autopay=autopay)
-            if save_card or is_saved_card:
-                customer = self.payment_method.customer
+                if save_card or is_saved_card:
+                    customer = self.payment_method.customer
+
+                payment_record = PaymentRecord.create_record(
+                    self.payment_method, None, amount
+                )
+                self.update_credits(payment_record)
+
+                charge = self.create_charge(amount, card=card, customer=customer)
 
-            charge = self.create_charge(amount, card=card, customer=customer)
+            payment_record.transaction_id = charge.id
+            payment_record.save()
             self.update_payment_information(billing_account)
```

Comment:
yes - my intention is that the payment record is committed if and only if the charge is processed via stripe without error.  So I'm relying on the stripe command being at the end of the ```atomic``` block.  If any of those 3 lines error for unforeseen reasons, I don't want the payment record to be rolled back when the credit card has already been charged.